### PR TITLE
Add tag coloring helpers to tag_resolver

### PR DESCRIPTION
**Pull request by michal-lightly**
## What has changed and why?

Create two bulk lookup helpers in `tag_resolver`. They will be used for embedding coloring:

- `get_names_by_ids(session, tag_ids)` — returns `{tag_id: name}` for a list of tag IDs.
- `get_tags_by_sample(session, tag_ids)` — returns `{sample_id: {tag_id, ...}}` for sample-tag membership lookups.

## How has it been tested?

New tests.

## Did you update [CHANGELOG.md](../CHANGELOG.md)?

- [x] Not needed (internal change)

## Summary by CodeRabbit

* **New Features**
  * Added batch query helpers to efficiently fetch tag names and to retrieve tag associations per sample, improving performance for tag-related lookups.

* **Tests**
  * Added comprehensive unit tests covering normal and edge cases for the new tag lookup behaviors (empty inputs, non-existent IDs, and membership mappings).

[Review Change Stack](https://app.coderabbit.ai/change-stack/lightly-ai/lightly-studio/pull/1172?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: lightly_studio/src/lightly_studio/resolvers/tag_resolver.py [hunk 1]
```diff
@@ -203,6 +204,32 @@ def remove_sample_ids_from_tag_id(
...
+    """Return ``{sample_id: {tag_id, ...}}`` for the requested tags."""
+    if not tag_ids:
+        return {}
+    stmt = select(SampleTagLinkTable.sample_id, SampleTagLinkTable.tag_id).where(
+        col(SampleTagLinkTable.tag_id).in_(tag_ids)
+    )
+    result: dict[UUID, set[UUID]] = {}
+    for sample_id, tag_id in session.exec(stmt).all():
+        assert sample_id is not None
+        assert tag_id is not None
```
[the remark this answers is not in the document]
Comment (michal-lightly) [reply lightly_studio/src/lightly_studio/resolvers/tag_resolver.py:228]: The asserts are there just for mypy to pass. The fields cannot be None.

**Review by LeonardoRosaa — APPROVED** [review]
